src/main/java/duke/FileSaver.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * This TaskList class handles the logic of adding and deleting tasks of Duke
+ * 
+ * @param file Task file to remember the tasks
+ * 
+ * @author WangYihe
+ * @author E0424695
+ */
+
+package duke.fileSaver;
+
+import duke.ui.Ui;
+import duke.exception.DukeException;
+import duke.task.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.List;
+
+import java.util.Scanner;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+public class FileSaver {
+    public File file;
+    
+    public FileSaver(String folderName, String fileName) {
+        file = new File(folderName, fileName);
+    }
+
+    public void createFile() throws DukeException {
+        try {
+            if (!file.getParentFile().exists()) {
+                file.getParentFile().mkdirs();
+                System.out.println("CREATED_FOLDER");
+            }
+            if (!file.exists()) {
+                file.createNewFile();
+                System.out.println("CREATED_SAVE_FILE");
+            }
+        } catch (IOException e) {
+            //TODO: handle exception
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+    public void addTask(TaskList task, Task t, String isDone) {
+        if (isDone.trim().equals("1")) {
+            t.markAsDone();
+        } 
+        task.add(t);
+    }
+
+    public static boolean isDateFormat(String date) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-mm-dd");
+        dateFormat.setLenient(false);
+        try {
+            dateFormat.parse(date.trim());
+        } catch (ParseException e) {
+            //TODO: handle exception
+            return false;
+        }
+        return true;
+    }
+
+    public void parseTask(TaskList task, String info) throws IndexOutOfBoundsException{
```

Comment:
If it is a "TaskList", perhaps can rename the variable name to "tasks"? 🤔  (same applies to the rest methods)